### PR TITLE
Add support to SyntaxTree 6 and Mermaid.js

### PR DESCRIPTION
**Pull request by wenderjean**
Related to https://github.com/ruby-syntax-tree/syntax_tree/issues/287

## Result
image
image
image

## Tasks
- [x] Update SyntaxTree
- [x] Add suport to mermaid.js
- [ ] Visualize CFG/DFG
- [ ] Visualize actual execution

## Improvements
- [ ] It would be nice to have zoom and drag and drop in graph visualization, similar to https://mermaid.live/
- [ ] Put some throttle when "source-changed" is dispatched to avoid too much rerendering

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -16,70 +16,22 @@ <h1>Syntax Tree</h1>
...
-        header += file.readline if header.start_with?("#!")
-        Ripper.new(header).tap(&:parse).encoding
-      end
-
-    File.read(filepath, encoding: encoding)
-  end
-end
-</textarea>
+      <textarea id="editor">
+SyntaxTree::Binary[
```
Comment (wenderjean) [inline docs/index.html]: The reason I'm changing the default example here is the fact that mermaid.js has a text limit it is able to turn into a Graph.

File: Rakefile [hunk 2]
```diff
@@ -19,7 +19,7 @@ file "head-wasm32-unknown-wasi-full-js" do
   version = JSON.parse(File.read("package.json"))["dependencies"]["ruby-head-wasm-wasi"][1..]
   filename = "ruby-head-wasm32-unknown-wasi-full-js.tar.gz"
 
-  `curl -LO https://github.com/ruby/ruby.wasm/releases/download/ruby-head-wasm-wasi-#{version}/#{filename}`
+  `curl -LO https://github.com/ruby/ruby.wasm/releases/download/ruby-head-wasm-wasi-#{version}/ruby-head-wasm32-unknown-wasi-full-js.tar.gz`
```
Comment (kddnewton) [inline Rakefile]: I think this line is equivalent and can be reverted

File: docs/index.html [hunk 3]
```diff
@@ -16,70 +16,22 @@ <h1>Syntax Tree</h1>
         <span><button type="button" id="format" disabled>Format</button></span>
 
         <div class="toggles">
-          <span><button type="button" value="prettyPrint" disabled>AST</button></span>
-          <span><button type="button" value="disasm" disabled>ISEQ</button></span>
+          <select>
```
Comment (kddnewton) [inline docs/index.html]: Let's start the `select` as disabled until the imports resolve. Also we should add an aria-label here for accessibility

File: docs/index.html (hunk 1 above)
Comment (kddnewton) [reply docs/index.html]: That's fine, but let's do one that is less confusing. Even just `1 + 2 * 3` would be fine.

File: src/createRuby.ts [hunk 4]
```diff
@@ -66,10 +66,13 @@ export default async function createRuby() {
   // files to make it work. I'm not sure why I need to explicitly require
   // did_you_mean here, but it doesn't work without it.
   ruby.eval(`
+    require "rubygems"
```
Comment (kddnewton) [inline src/createRuby.ts:69]: Is this require and the `pp` one necessary? I hope they aren't?

**Review by kddnewton — CHANGES_REQUESTED** [review]
Nice work @wenderjean this is a really great start!

File: src/createRuby.ts (hunk 4 above)
Comment (wenderjean) [reply src/createRuby.ts:69]: Unfortunately, they are. Without them, we get `unutilized Gem lib/syntax_tree/formatter.rb:10` and `unutilized PP` in the browser.

File: Rakefile (hunk 2 above)
Comment (wenderjean) [reply Rakefile]: You're right, It was just a missing test.

**Comment by kddnewton** [comment]
@wenderjean This is looking really good! Can we get this merged without support for CFGs? I'm happy to add that later. I just wasn't sure if you were ready for this to be reviewed or not.

**Comment by wenderjean** [comment]
@kddnewton Sure, I didn't have a chance to add CFGs yet but this PR is ready for review/merge.

**Review by kddnewton — APPROVED** [review]

**Pull request merged**